code/modules/mob/living/carbon/xenomorph/evolution.dm

```diff
@@ -316,8 +316,10 @@
 		if(death_timer)
 			to_chat(src, span_warning("The hivemind is still recovering from the last [initial(new_caste.display_name)]'s death. We must wait [DisplayTimeText(timeleft(death_timer))] before we can evolve."))
 			return FALSE
+
 	var/maximum_active_caste = new_caste.maximum_active_caste
-	if(maximum_active_caste != INFINITY && maximum_active_caste <= length(hive.xenos_by_typepath[new_caste_type]))
+	var/active_caste = length(hive.get_all_caste_members(new_caste.type))
+	if(maximum_active_caste != INFINITY && maximum_active_caste <= active_caste)
 		to_chat(src, span_warning("There is already a [initial(new_caste.display_name)] in the hive. We must wait for it to die."))
```

Comment:
Should be updated so it doesn't report a caste there could be zero of.

Reply:
fix

Reply:
did you fix this
Also I like that you're making sure it's not infinity before checking to see if it's smaller than something else
in case that other thing is a bigger infinity